Makefile

```diff
@@ -0,0 +1,40 @@
+MDBOOK_VERSION := v0.4.5
+SOURCE_PATH := docs/user-guide
+CONTAINER_RUNTIME := docker
+IMAGE_NAME := quay.io/metal3-io/mdbook
+HOST_PORT := 3000
+
+## ----------------------
+## Documentation tooling
+## ----------------------
+
+.PHONY: docker
+docker: # Build mdbook local container image
+	$(CONTAINER_RUNTIME) build . -t $(IMAGE_NAME):$(MDBOOK_VERSION)
+
+.PHONY: build
+build: # Build the user guide
+	$(CONTAINER_RUNTIME) run \
+	--rm -it --name metal3 \
+	-v "$$(pwd):/workdir" \
+	-u "$$(id -u):$$(id -g)" \
+	$(IMAGE_NAME):$(MDBOOK_VERSION) \
+	mdbook build $(SOURCE_PATH)
```

Comment:
This PR doesn't build for me

```
$ CONTAINER_RUNTIME=podman make -b build
podman run \
--rm -it --name metal3 \
-v "$(pwd):/workdir" \
-u "$(id -u):$(id -g)" \
quay.io/metal3-io/mdbook:v0.4.5 \
mdbook build docs/user-guide
2021-01-11 14:41:49 [ERROR] (mdbook::utils): Error: Couldn't open SUMMARY.md
2021-01-11 14:41:49 [ERROR] (mdbook::utils): 	Caused By: No such file or directory (os error 2)
make: *** [Makefile:17: build] Error 101
```

I don't see a `SUMMARY.md` in the branch for the PR. Does it need to be rebased?

Reply:
Sorry I had to mention in the description that it's normal, because we will have the `SUMMARY.md` after https://github.com/metal3-io/metal3-docs/pull/154 is merged. Because that's the PR that introduces `SUMMARY.md`

Reply:
If you want to try it out, you can copy [/docs/user-guide](https://github.com/Nordix/metal3-docs/tree/user-guide/feruz/docs/user-guide) directory which includes the `SUMMARY.md`

Reply:
#154 merged, and this branch now has conflicts. I'll try again after you rebase to fix those conflicts and fix the variable definition.

Reply:
Now I get

```
$ CONTAINER_RUNTIME=podman make build
podman run \
--rm -it --name metal3 \
-v "$(pwd):/workdir" \
-u "$(id -u):$(id -g)" \
fmuyassarov/mdbook:v0.4.5 \
mdbook build docs/user-guide
2021-01-12 15:24:56 [INFO] (mdbook::book): Book building has started
2021-01-12 15:24:56 [INFO] (mdbook::book): Running the html backend
2021-01-12 15:24:56 [ERROR] (mdbook::utils): Error: Rendering failed
2021-01-12 15:24:56 [ERROR] (mdbook::utils): 	Caused By: Unable to remove stale HTML output
2021-01-12 15:24:56 [ERROR] (mdbook::utils): 	Caused By: Permission denied (os error 13)
make: *** [Makefile:35: build] Error 101
```

Removing the `-u` option makes the build work. Is that option required for docker?

Reply:
That's weird. It works fine for me though
```
fmuyassarov@est:~/final/metal3-docs$ make build
docker run \
--rm -it --name metal3 \
-v "$(pwd):/workdir" \
-u "$(id -u):$(id -g)" \
fmuyassarov/mdbook:v0.4.5 \
mdbook build docs/user-guide
2021-01-12 16:11:27 [INFO] (mdbook::book): Book building has started
2021-01-12 16:11:27 [INFO] (mdbook::book): Running the html backend
```
If we don't use -u, then the generated book directory will have root:root ownership
```
~metal3-docs/docs/user-guide$ ls -lah
total 24K
drwxrwxr-x 5 fmuyassarov fmuyassarov 4.0K Jan 12 18:12 .
drwxrwxr-x 3 fmuyassarov fmuyassarov 4.0K Jan 11 22:24 ..
drwxr-xr-x 6 root        root        4.0K Jan 12 18:12 book
-rw-rw-r-- 1 fmuyassarov fmuyassarov  207 Jan 11 17:31 book.toml
drwxrwxr-x 3 fmuyassarov fmuyassarov 4.0K Jan 11 17:31 src
drwxrwxr-x 2 fmuyassarov fmuyassarov 4.0K Jan 11 17:31 theme
```
So, with `-u` we could make sure that the `book` directory is under `${USER}:${USER}`, which would be needed for `make serve`. But now, I think it would make sense to add `sudo` for docker/podman commands, so that we don't have to worry even if `book` directory is `${USER}:${USER}', because `make serve` also will run as root. 

@dhellmann can you please run `make clean & make build` ? I have removed `-u` flag.

Reply:
sudo really shouldn't be needed. Do we need a script to wrap the call to mdbook to change the ownership of the output?

Reply:
Actually no. Mdbook doesn't set the root ownership. I think it was misconfiguration in the docker flag. Agree with you. I am going to remove it now. Even without the sudo, -u I'm able to run successfully. Please let me know if that's the same for you too with podman.

Reply:
Part of the difference may be in how podman and docker run the container. Maybe the `sudo` should be part of the default value of `CONTAINER_RUNTIME` so when someone overrides it with `podman` the `sudo` can be dropped.

Reply:
True. By default docker requires you to run with sudo, which isn't the case for podman. So, I thought adding sudo won't even users running podman.

Reply:
> Part of the difference may be in how podman and docker run the container. Maybe the `sudo` should be part of the default value of `CONTAINER_RUNTIME` so when someone overrides it with `podman` the `sudo` can be dropped.

Yes, I think this will make sense.

Reply:
Updated it.